feat(example): init project

TFK70 opened the pull request:
affects: @example/app

ISSUES CLOSED: #83

Commit:
feat(example): init project

affects: @example/app

ISSUES CLOSED: #83

Commit:
feat(example): init input

affects: @example/app, @ex-ui/input

Commit:
refactor(example): finalize input

affects: @example/app, @ex-ui/input

Commit:
feat(example): init theme

affects: @ex-ui/theme, @web-diary/app, @practice/web-diary

gochicus committed:
feat(example): example markup added

affects: @fragments/controls, @ex-ui/button, @ex-ui/color-picker, @ex-ui/layout, @ex-ui/text

gochicus committed:
feat(example): controls fragment added

affects: @example/app, @fragments/controls, @ex-ui/button, @ex-ui/color-picker, @ex-ui/input,
@ex-ui/layout, @ex-ui/text

ISSUES CLOSED: #83

gochicus committed:
WIP(example): view fragment, logic added

affects: @example/app, @fragments/controls, @fragments/example-build, @fragments/view,
@store/color-data, @store/input-data, @ex-ui/button, @ex-ui/color-picker, @ex-ui/layout,
@practice/food-delivery

ISSUES CLOSED: #83

gochicus committed:
feat(example): addon functionality added

affects: @example/app, @fragments/controls, @fragments/example-build, @fragments/view,
@ex-ui/button, @ex-ui/color-picker, @ex-ui/input, @ex-ui/layout, @ex-ui/text, @ex-ui/theme

ISSUES CLOSED: #83

gochicus committed:
refactor(example): input functionality changed

affects: @fragments/controls, @ex-ui/input

ISSUES CLOSED: #83

gochicus committed:
feat(example): background component added

affects: @example/app, @fragments/constructor, @fragments/controls, @fragments/view,
@ex-ui/background, @ex-ui/button

ISSUES CLOSED: #83

TFK70 commented on practice/example/fragments/view/src/View.tsx:
```diff
@@ -9,7 +10,7 @@ export const View = () => {
   const [data] = useInputData()
   return (
-    <Box backgroundColor={value} justifyContent='center' width='100%'>
+    <Background backgroundColor={value} justifyContent='center' width='100%'>
```
background по умолчанию должен занимать всю ширину/высоту, это не нужно описывать снаружи

gochicus replied:
@TFK70 по умолчанию не работает . Отбой, ошибся

TFK70 commented on practice/example/ui/background/src/Background.tsx:
```diff
@@ -0,0 +1,10 @@
+Background.defaultProps = {
+  backgroundColor: 'silver',
+  display: 'flex',
+  boxSizing: 'border-box',
```
Зачем?

TFK70 commented on practice/example/app/src/App.tsx:
```diff
@@ -13,7 +13,7 @@ const App = () => {
       <ColorDataProvider value={[value, setValue]}>
         <ThemeProvider>
-          <ExampleBuild />
+          <Constructor />
```
У тебя там находится твое приложение, а не конструктор

Т.е. представь вёрстку:

```javascript
<Header /> // шапка
<AuthorizationForm /> // форма авторизации
<Constructor />  // ???конструктор???
<Footer /> // подвал
```

gochicus replied:
@TFK70 просто в Example переименовать?

TFK70 replied:
@gochicus ну в данном случае можно и Example, мне трудно сказать какую проблему решает наше приложение

gochicus replied:
@TFK70 может лучше будет ExampleApp чтобы дважды example не было?

TFK70 reviewed (commented):
И у тебя чеки упали

gochicus committed:
refactor(example): component name changed

affects: @example/app, @fragments/example-app, @fragments/view, @ex-ui/background

ISSUES CLOSED: #83

TFK70 commented on practice/example/ui/color-picker/src/ColorPicker.tsx:
```diff
@@ -0,0 +1,33 @@
+import { useColorData }        from '@store/color-data'
+
+export const ColorPicker = ({ colors }) => {
+  const [, setValue] = useColorData()
```
Ну вот зачем это здесь?
ColorPicker - просто независимый компонент который мапится по массиву со строками. Переделай его по тому же принципу: у тебя должно быть 2 работающих независимо друг от друга ColorPicker'а

Это было бы допустимо будь твой ColorPicker каким-нибудь фрагментом (но выносить 3 кнопки в отдельный фрагмент - странно), компоненты ui отвесают только за отрисовку и все. Их ни в коем случае нельзя наделять дополнительной логикой как сейчас

gochicus replied:
@TFK70 А зачем два колорпикера?

TFK70 replied:
@gochicus чтобы проверить что твой ui компонент корректно спроектирован
Так же как мы с инпутом делали

gochicus replied:
@TFK70 сформулируй что нужно сделать, два колорпикера работают как есть.

TFK70 commented on practice/example/ui/input/src/appearance/appearance.styles.ts:
```diff
@@ -0,0 +1,39 @@
+})
+
+export const createBorderFocusStyles = (color): styleFn => () => ({
+  '&:hover': {
```
createBorderFoucsStyles > '&:hover'

Нужно переименовать

gochicus commented:
@TFK70 Если передавать всю логику onClick в ColorPicker, который расположен в Controls, каким образом передать туда значения текущего цвета кнопки, не используя состояния в ColorPicker?

gochicus committed:
refactor(example): colorPicker component refactored

affects: @fragments/controls, @ex-ui/color-picker, @ex-ui/input, @practice/food-delivery,
@web-diary/app, @practice/web-diary

ISSUES CLOSED: #83

TFK70 commented on practice/example/fragments/controls/src/Controls.tsx:
```diff
@@ -24,7 +27,7 @@ export const Controls = () => {
           }}
         />
-        <ColorPicker colors={['red', 'green', 'blue']} />
+        <ColorPicker updateState={updateState} colors={['red', 'green', 'blue']} />
```
В точку (ну почти), я все-таки немного не понимаю зачем было оборачивать это все в какой-то updateState когда ты с тем же успехом мог прокинуть просто setValue

TFK70 replied:
Это поправь и ок

gochicus committed:
refactor(example): controls component refactored

affects: @fragments/controls

ISSUES CLOSED: #83

gochicus committed:
refactor(example): button styles logic refactored

affects: @fragments/controls, @ex-ui/button, @ex-ui/color-picker

ISSUES CLOSED: #83
